cadcdata/cadcdata/storageinv.py

```diff
@@ -560,35 +592,61 @@ def cadcput(self, id, src, replace=False, file_type=None,
                 result = self._cadc_client.post(url, headers=headers)
                 result.raise_for_status()
                 duration = time.time() - start
-                logger.debug('Updated metadata for identifier {} in {} ms'.
-                             format(id, duration))
+                logger.info('Updated metadata for identifier {} in {} ms'.
+                            format(id, duration))
                 return
             logger.debug('PUT to URL {}'.format(url))
             try:
                 start = time.time()
-                with open(src, 'rb') as f:
-                    self._cadc_client.put(url, headers=headers, data=f)
+                with util.Md5File(src, 'rb') as reader:
+                    self._cadc_client.put(url, headers=headers, data=reader)
                 duration = time.time() - start
                 stat_info = os.stat(src)
-                logger.debug(
+                if md5_checksum and (md5_checksum != reader.md5_checksum):
+                    raise RuntimeError(
```

Comment:
I think that requiring users of this method to provide an md5sum, and then calculating the md5sum to compare against the provided value, is going to end up in doing something twice that can be an expensive operation. My _opinion_ is that the client should do the calculation, and the user of the function should ask for this behaviour via a flag.

If there are other reasons (e.g. for future needs that I'm not aware of) for this implementation, ignore this comment.

Reply:
Ah! Forgot to add the info to the headers of the initial request. Fixed now.
The client calculates the md5 of the bytes as it sends them so that overhead is minimal. However, sending the md5 with the initial request allows the service to discard a corrupted file right away in case the client does not re-transmit it so it's a good thing to have (but expensive to pre-compute). The caller might need to calculate this for other purposes so having a way to pass the info to `cadcput` could be useful.
This might all change with PUT transactions but I'm not sure about it. It an `rc` version so we can drop it later if that's the case.